core/src/main/java/de/danoeh/antennapod/core/service/PlayerWidgetJobService.java

```diff
@@ -95,6 +116,20 @@ private void updateViews() {
         if (media != null) {
             views.setOnClickPendingIntent(R.id.layout_left, startMediaplayer);
 
+            try {
+                Bitmap icon = null;
+                int iconSize = getResources().getDimensionPixelSize(android.R.dimen.app_icon_size);
+                icon = Glide.with(PlayerWidgetJobService.this)
+                        .asBitmap()
+                        .load(media.getImageLocation())
+                        .apply(RequestOptions.diskCacheStrategyOf(ApGlideSettings.AP_DISK_CACHE_STRATEGY))
+                        .submit(iconSize, iconSize)
+                        .get();
```

Comment:
If you use something like `.get(500, TimeUnit.MILLISECONDS)`, you can make sure that loading the icon does not cause "Application not responding" errors. I think 500ms are a reasonable maximum duration because the widget is updated every 1 second.